packages/ens-utils/src/domain.ts

```diff
@@ -0,0 +1,340 @@
+  labelName: string;
+  /** keccak256 hash of the label */
+  labelHash: string;
+  unwrappedTokenId: bigint;
+  wrappedTokenId: bigint;
+};
+
+export type DomainCard = {
+  name: ENSName;
+
+  /**
+   * A reference to the NFT associated with `name`.
+   *
+   * null if and only if one or more of the following are true:
+   * 1. name is not normalized
+   * 2. name is not currently minted (name is on primary market, not secondary market) and the name is not currently expired in grace period
+   * 3. we don't know a strategy to generate a NFTRef for the name on the specified chain (ex: name is associated with an unknown registrar)
+   */
+  nft: NFTRef | null;
+  parsedName: DomainName;
+  registration: Registration;
+  /** Stringified JSON object with debug information about the name generator */
+  nameGeneratorMetadata: string | null;
+  /** Whether the domain is on watchlist */
+  onWatchlist: boolean;
+  ownerAddress: `0x${string}` | null;
+  managerAddress: `0x${string}` | null;
+  /** Former owner address is only set when the domain is in Grace Period */
+  formerOwnerAddress: `0x${string}` | null;
+  /** Former manager address is only set when the domain is in Grace Period */
+  formerManagerAddress: `0x${string}` | null;
+};
+
+/**
+ * Returns the expiration timestamp of a domain
+ * @param domainRegistration Registration object from domain
+ * @returns Timestamp | null
+ */
+export function domainExpirationTimestamp(
+  domainRegistration: Registration,
+): Timestamp | null {
+  if (domainRegistration.expirationTimestamp) {
+    return domainRegistration.expirationTimestamp;
+  }
+  return null;
+}
+
+/**
+ * Returns the release timestamp of a domain, which is 90 days after expiration when the Grace Period ends
+ * @param domainRegistration Registration object from domain
+ * @returns Timestamp | null
+ */
+export function domainReleaseTimestamp(
+  domainRegistration: Registration,
+): Timestamp | null {
+  const expirationTimestamp = domainExpirationTimestamp(domainRegistration);
+  if (expirationTimestamp === null) return null;
+
+  const releaseTimestamp = addSeconds(expirationTimestamp, GRACE_PERIOD);
+  return releaseTimestamp;
+}
+
+/* 
+  Below enum options differ based on domain's owner
+  and on its secondary marketplace status:
+  If domain has no owner: noOwner;
+  If domain has an owner but user is not the owner: notOwner;
+  If user is owner of the domain and domain is in Grace Period: formerOwner;
+  If user is owner of the domain and domain is not in Grace Period: activeOwner;
+*/
+export enum UserOwnershipOfDomain {
+  noOwner = "noOwner",
+  notOwner = "notOwner",
+  formerOwner = "formerOwner",
+  activeOwner = "activeOwner",
+}
+
+/**
+ * Returns the ownership status of a domain in comparison to the current user's address
+ * @param domain Domain that is being checked
+ * @param currentUserAddress Address of the current user.
+ * @returns UserOwnershipOfDomain
+ */
+export const getCurrentUserOwnership = (
```

Comment:
It was built this way so integrators do not have to deal with wether they have the information yet or not, alone.

By accepting these nulls we the function says "I understand sometimes you won't have the currentUserAddress or the domain card yet and that is OK"

I understand there are different ways for building this so I am happy to hear what you think and know about a good approach. 

The function is, yet, resolving things nicely by saying: "If no domain is informed, I will say the Ownership is no owner... If no currentUser is informed, I will say the information regarding the domain: if it has owner, notOwner is my response since currentUser is not informed, but, if it has no owner, noOwner is my response"

Reply:
@FrancoAguzzi Ok thanks. I can appreciate that direction. We need to document ideas like this when we document the function.